### PR TITLE
feat: Add optional full_history boolean flag

### PR DESCRIPTION
**Pull request by JeroenKnoops**
Add optional `full_history` argument.

When set to `true` the repository is being checked out with the complete history.
If set to `false` (the default), the repository is being checked out with `depth=1`

Default value is `false`.

cc @vanGastelJS

### CODE REVIEW

**Comment by vanGastelJS** [comment]
@npalm: Could you please review this pull request as I need this for correct builds using build automation. Github does not (yet) support git submodules.

File: action.yml [hunk 1]
```diff
@@ -16,6 +16,10 @@ inputs:
     description: base dir for cloning actions
     required: false
     default: '.actions'
+  full_checkout:
+    description: Full Checkout of the repository
```
Comment (npalm) [inline action.yml:20]:
```suggestion
    description: By defaullt a shallow clone with depth 1 is created. Set to `true` to clone the repo will a full history.
```

**Review by npalm — CHANGES_REQUESTED** [review]
@vanGastelJS just one small suggestion. Thanks for the contribution.

**Comment by vanGastelJS** [comment]
@npalm: Could you please review the updated pull request? And if ok, could you also release this feature so I can use it in our automated builds?

**Review by npalm — APPROVED** [review]

**Pull request merged**